test/Microsoft.AspNet.Mvc.IntegrationTests/DictionaryModelBinderIntegrationTest.cs

```diff
@@ -134,6 +137,7 @@ public class DictionaryModelBinderIntegrationTest
         [Theory]
         [InlineData("?[key0]=10")]
         [InlineData("?[0].Key=key0&[0].Value=10")]
+        [InlineData("?index=low&[low].Key=key0&[low].Value=10")]
```

Comment:
This is an integration test.

Because the explicit index syntax is usable in scenarios using a POCO `TValue`, could test with a few data annotation attributes in place. That would ensure validation works correctly in this case. Is that what you're looking for?